pandas/core/indexes/base.py

```diff
@@ -1339,6 +1342,237 @@ def rename(self, name, inplace=False):
         """
         return self.set_names([name], inplace=inplace)
 
+    def replace(
+        self,
+        to_replace=None,
+        value=None,
+        inplace=False,
+        limit=None,
+        regex=False,
+        method="pad",
+    ):
+        if inplace:
+            raise TypeError("Index can't be updated inplace.")
+
+        if not is_bool(regex) and to_replace is not None:
+            raise AssertionError("'to_replace' must be 'None' if 'regex' is not a bool")
+
+        if value is None:
+            if isinstance(to_replace, (tuple, list)):
+                fill_f = missing.get_fill_func(method)
+                mask = missing.mask_missing(self.values, to_replace)
+                new_index = fill_f(self.values, limit=limit, mask=mask)
+                new_index = new_index.astype(self.dtype)
+
+                return self._constructor(new_index)
+
+            if not is_dict_like(to_replace) and not is_dict_like(regex):
+                raise NotImplementedError(
+                    "This is implemented in NDFrame.replace(). However,"
+                    "not clear if we should include this in the API."
+                    "See issue 5319 and PR 5600. But also note that this"
+                    "use is not mentioned in the docs."
+                )
+
+            if not is_dict_like(to_replace):
+                if not is_dict_like(regex):
+                    raise TypeError(
+                        'If "to_replace" and "value" are both None '
+                        'and "to_replace" is not a list, then '
+                        "regex must be a mapping"
+                    )
+                to_replace = regex
+                regex = True
+
+            items = list(to_replace.items())
+            keys, values = zip(*items) if items else ([], [])
+
+            to_replace, value = keys, values
+
+            return self.replace(
+                to_replace=to_replace,
+                value=value,
+                inplace=inplace,
+                limit=limit,
+                regex=regex,
+            )
+
+        else:
+            if is_dict_like(to_replace):
+                raise TypeError("If `to_replace` is a dict, `value` should be None.")
```

Comment:
Replace backticks with single quotes.

(To he consistent with the error messages)